### PR TITLE
Stop Conditions: queue_length and empty

### PR DESCRIPTION
**Pull request by rphh**
This includes tests and required fixes for stop condition translator

### CODE REVIEW

File: sdl2promela/stop_condition/translator.py [hunk 1]
```diff
@@ -1018,6 +1022,40 @@ def _generate_entry_loop() -> promela.Do:
...
+    expressions = []
+
+    # Collect all expressions
+    for always in input_model.always_statements:
+        expressions.append(_generate(context, always.expression))
+
+    for never in input_model.never_statements:
+        expressions.append(_generate(context, model.NotExpression(never.expression)))
+
+    expression = expressions[0]
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:1037]: Why first expression is ignored?

File: sdl2promela/stop_condition/translator.py [hunk 2]
```diff
@@ -1018,6 +1022,40 @@ def _generate_entry_loop() -> promela.Do:
     )
 
 
+def _generate_else_alternative(
+    input_model: model.StopConditionModel, context: GenerateContext
+) -> promela.Alternative:
+    expressions = []
+
+    # Collect all expressions
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:1030]: all invariant? There is no filter out and eventualy...

File: sdl2promela/stop_condition/translator.py (hunk 1 above)
Comment (Lurkerpas) [reply sdl2promela/stop_condition/translator.py:1037]: OK, never mind, my mistake

File: sdl2promela/stop_condition/translator.py [hunk 3]
```diff
@@ -1042,12 +1080,16 @@ def _build_simple_never_claim(
             _generate_filter_out_alternative(input_model.filter_out_statements, context)
         )
     else:
-        do_loop_builder.withAlternative(_generate_true_alternative("start"))
+        do_loop_builder.withAlternative(
+            _generate_else_alternative(input_model, context)
+        )
+        # do_loop_builder.withAlternative(_generate_true_alternative("system_inited"))
```
Comment (Lurkerpas) [inline sdl2promela/stop_condition/translator.py:1086]: Why is this commented out?

File: tests/integration/stop_condition_empty_len/Makefile [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	cp ${RES_DIR}/tmc/env_inlines.pml ${TEST_DIR}/env_inlines.pml
+	cd ${TEST_DIR} && \
+	${SPIN} -a -m system.pml
+	$(CC) -o ${TEST_DIR}/pan ${TEST_DIR}/pan.c $(CFLAGS) -DNFAIR=3
+	cd ${TEST_DIR} && \
+	./pan -a -f -n -m1000000 > system.output
+	grep -q "errors: 1" ${TEST_DIR}/system.output
+	${DELETE_TEST_DIR}
+
+# In this test the model checker shouln't find an error.
```
Comment (Lurkerpas) [inline tests/integration/stop_condition_empty_len/Makefile:77]: shouldn't

File: tests/integration/stop_condition_empty_len/Makefile [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+# These tests are created based on the tests supplied with custom Spin version
+# used by N7 Space (see github.com/n7space/Spin). Their purpose is to validate
+# whether the generated Promela code actually works inside Spin.
+# As sdl2promela generates only inlines (representing the SDL logic) which need
+# external data and process definitions (similarly to opengeode generating
+# only C functions, which need type declarations generated by asn1scc and
+# glue code generated by Kazoo), they are wrapped inside "host" Promela
+# files
+
+# In this test the model checker shouln't find an error.
```
Comment (Lurkerpas) [inline tests/integration/stop_condition_empty_len/Makefile:41]: shouldn't

File: tests/unit/translator/resources/join.pml [hunk 6]
```diff
@@ -11,18 +11,15 @@ inline Complexsdl_0_transition(id)
...
-    ;
-    jump: ;
+    jump:
     Complexsdl_0_RI_0_response(global_state.complexsdl.tmp);
     global_state.complexsdl.state = Complexsdl_States_wait;
     transition_id = -1;
     goto next_transition;
-    ;
-    next_transition: ;
+    next_transition:
```
Comment (Lurkerpas) [inline tests/unit/translator/resources/join.pml:22]: Is it ok if there is no instruction after the label?

Comment (rphh) [reply tests/unit/translator/resources/join.pml:22]: Yes, this construction is accepted by spin.

File: sdl2promela/stop_condition/translator.py (hunk 2 above)
Comment (rphh) [reply sdl2promela/stop_condition/translator.py]: This should be removed

File: tests/integration/stop_condition_empty_len/Makefile (hunk 4 above)
Comment (rphh) [reply tests/integration/stop_condition_empty_len/Makefile]: fixed

File: tests/integration/stop_condition_empty_len/Makefile (hunk 5 above)
Comment (rphh) [reply tests/integration/stop_condition_empty_len/Makefile]: fixed

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**